### PR TITLE
Adjust to deprecations for graphql_core v.3.2

### PR DESCRIPTION
**Pull request by kristjanvalur**
the `format_error()` function to turn `GraphQLError` into `GraphQLFormattedError` is deprecated
for `graphql-core` 3.2 and will be removed for version 3.3.

`graphql.subscription` module is deprecated, its contents moved to `graphql.execution`

No other deprecated features seem to be in use in strawberry.

Current version in strawberry is "~3.2.0"

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Stop using deprecated functionality for graphql-core 3.2


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to @kristjanvalur for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py [hunk 1]
```diff
@@ -250,7 +249,7 @@ async def get_result_source():
         # Handle initial validation errors
         if isinstance(result_source, GraphQLExecutionResult):
             assert result_source.errors
-            payload = [format_graphql_error(result_source.errors[0])]
+            payload = [err.formatted for err in result_source.errors]
```
Comment (DoctorJohn) [inline strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py:252]: Nice catch :+1:

**Review by DoctorJohn — APPROVED** [review]
Thanks for getting Strawberry ready for graphql-core 3.3 @kristjanvalur. Straightforward yet super welcome changes!

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Release type: patch
+
+Stop using functionality deprecated in graphql-core 3.2
```
Comment (patrick91) [inline RELEASE.md:3]:
```suggestion
This release removes some usage of deprecated functions from GraphQL-core.
```

**Review by patrick91 — APPROVED** [review]

**Pull request merged**